Server/VikingVault.Services/UserService.cs

```diff
@@ -13,23 +13,34 @@ namespace VikingVault.Services
     public class UserService : IUserService
     {
         private readonly VikingVaultDbContext _dbContext;
-        public UserService(VikingVaultDbContext dbContext)
+        private readonly IBankAccountService _bankAccountService;
+
+        public UserService(VikingVaultDbContext dbContext, IBankAccountService bankAccountService)
         {
             _dbContext = dbContext;
+            _bankAccountService = bankAccountService;
         }
 
         public User CreateUser(User user)
         {
             user.Role = "user";
+            user.Password = PasswordEncryption.ComputeSha256Hash(user.Password);
+
             try
             {
-                user.Password = PasswordEncryption.ComputeSha256Hash(user.Password);
                 _dbContext.Add(user);
+
+                _bankAccountService.CreateBankAccount(this.CreateBankAccount(user, "ron"));
```

Comment:
I don't know the application's business logic, but it seems like this "ron", "eur".. strings could be replaced with enums